src/audio/smart_amp/smart_amp_generic.c

```diff
@@ -80,9 +80,9 @@ static void smart_amp_s24_ff_default(const struct comp_dev *dev,
 
 #if CONFIG_FORMAT_S32LE
 static void smart_amp_s32_ff_default(const struct comp_dev *dev,
-				     const struct audio_stream *source,
-				     const struct audio_stream *sink,
-				     const struct audio_stream *feedback,
+				     const struct audio_stream __sparse_cache *source,
+				     const struct audio_stream __sparse_cache *sink,
+				     const struct audio_stream __sparse_cache *feedback,
 				     uint32_t frames)
```

Comment:
Looks like this doesn't belong to this commit. Same with some other changes related to ```__sparse_cache```

Reply:
@dbaluta this very much does belong to it. Maybe the commit description isn't very clear, but that's exactly what you need to do to acquire and release buffers properly: with acquired buffers you need to use the proper sparse annotation, specifically this one.